src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2130 @@
+        The Trigger rate corresponds to the ebeam blanking frequency. As the delay
+        generator is operated "external", the trigger rate is a read-only value.
+        Called whenever an image arrives.
+        """
+        value = self.parent.DevParamGet(self.location, "Repetition Rate")
+        self._metadata[model.MD_TRIGGER_RATE] = value
+
+
+class StreakCamera(model.HwComponent):
+    """
+    Represents Hamamatsu readout camera for the streak unit.
+    Client to connect to HPD-TA software via RemoteEx.
+    """
+
+    def __init__(self, name, role, port, host, children=None, daemon=None, **kwargs):
+        """
+        Initializes the device.
+        :parameter host: (str) IP-adress or hostname
+        :parameter port: (int) port for sending/receiving commands
+        """
+        super(StreakCamera, self).__init__(name, role, daemon=daemon, **kwargs)
+
+        port_d = port + 1  # the port number to receive the image data
+        self.host = host
+        self.port = port
+        self.port_d = port_d
+
+        self._lock_command = threading.Lock()
+
+        # TODO start RemoteEx via SSH
+        # or TODO autostart of RemoteEx when turning on hamamatsu pc?
+
+        # connect to readout camera
+        try:
+            # initialize connection with RemoteEx client
+            self._commandport, self._dataport = self._openConnection()
+        except Exception:
+            raise logging.error("Failed to initialise Hamamatsu readout camera.")
+
+        # collect responses (error_code = 0-3,6-10) from commandport
+        self.queue_command_responses = Queue.Queue(maxsize=0)
+        # save messages (error_code = 4,5) from commandport
+        self.queue_img = Queue.Queue(maxsize=0)
+
+        self.should_listen = True  # used in readCommandResponse thread
+        self._waitForCorrectResponse = True  # used in sendCommand
+        self._getReadoutCamInfo = False  # nasty trick to get cam info
+
+        # start thread, which keeps reading the commandport response continuously
+        self._start_receiverThread()
+
+        # Note: start HPDTA after initializing queue and command and receiver treads
+        # but before image thread and initializing children!
+
+        # TODO check if already running....otherwise start multiple apps
+        # TODO  -> in acquisition mode it looks like it does not start a second app, but also does not report that
+        # TODO -> in processing mode it is possible to start multiple apps....
+        # TODO find out where to ask for acq or processing mode
+        # TODO is there a clever way for checking if app still running? Seems to be no command available to check
+        # TODO appEnd only works for the last opened window
+        # TODO want to check if we want to start app invisible (sVisible = False)
+
+        self.AppStart()  # start HPDTA software  # TODO comment out for testing in order to not start a new App
+
+        if children:
```

Comment:
Drop the `if`. And maybe add a:
`children = children or {}`

Reply:
In general I try to apply changes requested to both drivers :)